File: 1051-height-checker/main.cpp
class Solution {
public:
    int heightChecker(vector<int>& heights) {
        
        int n = heights.size();

        vector<int> expected(heights);
        sort(expected.begin(), expected.end());

        int count = 0;
        for (int i = 0; i < n; i++) {
            if (heights[i] != expected[i]) {
                count++;
            }
        }

        return count;

    }
};

File: 1051-height-checker/main.rs
impl Solution {
    pub fn height_checker(heights: Vec<i32>) -> i32 {
        
        let n: usize = heights.len();
        
        let mut expected: Vec<i32> = heights.clone();
        expected.sort();


        let mut count: i32 = 0;
        for i in 0..n {
            if heights[i] != expected[i] {
                count += 1;
            }
        }
        
        count
    }
}
